jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsAccountCache.kt

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.ServiceManager
+import software.amazon.awssdk.auth.credentials.AnonymousCredentialsProvider
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.sts.StsClient
+import software.aws.toolkits.core.ToolkitClientManager
+import software.aws.toolkits.core.credentials.ToolkitCredentialsChangeListener
+import software.aws.toolkits.core.credentials.ToolkitCredentialsProvider
+import software.aws.toolkits.core.utils.tryOrNull
+import software.aws.toolkits.jetbrains.core.credentials.CredentialManager
+import java.util.concurrent.ConcurrentHashMap
+
+interface AwsAccountCache {
+    /**
+     * Return the underlying AWS account for the given credential provider. Return null if the credential provider is invalid
+     */
+    fun awsAccount(credentialProvider: ToolkitCredentialsProvider): String?
+
+    companion object {
+        fun getInstance(): AwsAccountCache = ServiceManager.getService(AwsAccountCache::class.java)
+    }
+}
+
+class DefaultAwsAccountCache(private val stsClient: StsClient) : AwsAccountCache {
+    constructor() : this(ToolkitClientManager.createNewClient(
+        StsClient::class,
+        AwsSdkClient.getInstance().sdkHttpClient,
+        Region.US_EAST_1,
+        AnonymousCredentialsProvider.create(),
+        AwsClientManager.userAgent
+    ))
+
+    init {
+        ApplicationManager.getApplication().messageBus.connect().subscribe(
+            CredentialManager.CREDENTIALS_CHANGED,
+            object : ToolkitCredentialsChangeListener {
```

Comment:
could prolly just pass `this` and put the implementation methods on `DefaultAwsAccountCache`

Reply:
Done.